src/v2/Components/ConfirmPasswordModal/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button, Flex, Modal, space } from "@artsy/palette"
+import React from "react"
+import { useSystemContext } from "v2/Artsy"
+import { PasswordInput } from "v2/Components/PasswordInput"
+import { Form, Formik, FormikProps } from "formik"
+import { Error } from "v2/Components/Authentication/commonElements"
+import { ConfirmPassword } from "./Mutations/ConfirmPassword"
+import { ConfirmPasswordInput } from "v2/__generated__/ConfirmPasswordMutation.graphql"
+
+interface ConfirmPasswordModalProps {
+  onConfirm: (password: string, formikBag: FormikProps<any>) => void
+  onCancel: () => void
+  show: boolean
+  title?: string
+  inputMessage?: string
+}
+
+export const ConfirmPasswordModal: React.FC<ConfirmPasswordModalProps> = props => {
+  const { onCancel, onConfirm, show, title, inputMessage } = props
+  const { relayEnvironment } = useSystemContext()
+
+  const onSubmit = async (
+    { password }: ConfirmPasswordInput,
+    formikBag: FormikProps<any>
+  ) => {
+    formikBag.setStatus({ error: undefined })
+    try {
+      await ConfirmPassword(relayEnvironment, {
+        password,
+      })
+      onConfirm(password, formikBag)
+    } catch (err) {
+      formikBag.setStatus(err)
+    }
+  }
+
+  const onClickCancel = e => {
+    e.preventDefault()
+    onCancel()
+  }
+
+  return (
+    <Modal
+      show={show}
+      title={title || "Confirm your password"}
+      onClose={onCancel}
+    >
+      <Formik
+        initialValues={{
+          password: "",
+        }}
+        onSubmit={onSubmit}
+      >
+        {({
+          handleBlur,
+          handleChange,
+          handleSubmit,
+          isSubmitting,
+          status,
+          values,
+        }) => (
+          <Form
+            onSubmit={handleSubmit}
+            onKeyDown={e => {
+              if (e.key === "Enter") {
+                e.preventDefault()
+                handleSubmit()
+              }
+            }}
+          >
+            <PasswordInput
```

Comment:
**non-blocking**: Might be worth having some explanatory text above the input regarding the password confirmation.

@tinn might have some thoughts as he designed the original flow.

Reply:
@tinn open to suggestions! There is some explanation when opening the modal, in the form of a required field error:
<img width="590" alt="Screen Shot 2020-12-09 at 2 09 43 PM" src="https://user-images.githubusercontent.com/1497424/101675823-4d01d280-3a28-11eb-96ec-e1564f3e11e0.png">

Reply:
Channeling previous feedback I've received: it can be a jarring experience to be presented with an error state right away.

Might feel nicer to get some static explanatory text above the input and only use the error state/messaging when the form is submitted and we get an error back from Gravity.

Reply:
Taking a look. Thanks @eessex and @dblandin for tagging. Agree that the experience can be improved by showing the error state/messaging after form has been submitted. 

 @dblandin, it might be better for you to take another pass at what the copy text could be for this state. I was thinking something along the lines of: 

Header: Confirm security settings change 
Copy: Disabling two-factor authentication removes additional layers of security to your account. Enter your password to confirm this change. 
Actions: Cancel and Confirm

Reply:
We would need this copy for 3 states: Disable, Enable, and Edit.

Reply:
> Disabling two-factor authentication removes additional layers of security to your account.

It's a bit tricky to know whether 2FA would be completely disabled as a result of disabling a specific method because we currently allow multiple active methods. Here's what I'd suggest:

**Enable**

Header: Set up with app / Set up with text message
Copy: Confirm your password to continue.
Actions: Cancel / Confirm

**Edit**

Header: Set up with app / Set up with text message
Copy: Confirm your password to continue.
Actions: Cancel / Confirm

**Disable**

Header: Disable 2FA method?
Copy: Enter your password to confirm this change.
Actions: Cancel / Disable

I think signaling the user intent in the header is a good pattern to follow.

---

**non-blocking**: As an aside, the "Set up with text message" modal currently utilizes the multi-step wizard component. Might be an interesting experiment to see how a new initial step of "confirm password" might work vs a separate modal. Might simplify the callback / component interaction as bit.

Reply:
Looks good to me!